lib/services/local/generic/generic_test.go

```diff
@@ -264,3 +264,49 @@ func TestGenericCRUD(t *testing.T) {
 	require.Empty(t, nextToken)
 	require.Empty(t, out)
 }
+
+func TestMakeKey(t *testing.T) {
+	backendPrefix := "generic_prefix"
+	tests := []struct {
+		name     string
+		keyName  string
+		prefix   []string
+		expected []byte
+	}{
+		{
+			name:     "just a name",
+			keyName:  "key",
+			expected: backend.Key(backendPrefix, "key"),
+		},
+		{
+			name:     "name with parts",
+			keyName:  "key",
+			prefix:   []string{"part1", "part2"},
+			expected: backend.Key(backendPrefix, "part1", "part2", "key"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := context.Background()
+			memBackend, err := memory.New(memory.Config{
+				Context: ctx,
+				Clock:   clockwork.NewFakeClock(),
+			})
+			require.NoError(t, err)
+
+			service, err := NewService(&ServiceConfig[*testResource]{
+				Backend:       memBackend,
+				ResourceKind:  "generic resource",
+				PageLimit:     200,
+				BackendPrefix: backendPrefix,
+				UnmarshalFunc: unmarshalResource,
+				MarshalFunc:   marshalResource,
+			})
+			require.NoError(t, err)
+
+			key := service.MakeKey(test.keyName, test.prefix...)
```

Comment:
Suggestion: Implement `service.MakeKey` using a top-level `func MakeKey(prefix, name string, parts ...string)`, so we can test it without all the service setup?

If one just calls the other then visual inspection is enough. (Plus it gets exercised a bunch.)

That is, assuming we even want to keep the test?

Reply:
Seems reasonable to me.

Reply:
Oh actually, I remember now -- this is on purpose so that callers can use MakeKey and be assured that the associated backendPrefix is pushed into place. I'd rather leave that as is for that reason.

Reply:
They can still use s.MakeKey, right? I'm not sure I get the problem. If we can make MakeKey simpler I'm OK ditching the test, so this may all be a moot point.

Reply:
I don't think we can make `MakeKey` simpler, as per [this comment](https://github.com/gravitational/teleport/pull/30579#discussion_r1297511856). Let me know what you think.